### PR TITLE
Improved station search dialog

### PR DESCRIPTION
**Pull request by JasonLG1979**
Make better use of GtkTemplate
Moved search entry to headerbar
Added placeholder text to search entry
model is no longer populated if the search entry is empty

### CODE REVIEW

**Comment by TingPing** [comment]
> Moved search entry to headerbar

Seems a bit crowded to me.

**Comment by JasonLG1979** [comment]
> Seems a bit crowded to me.

What do you mean?

**Comment by JasonLG1979** [comment]
screenshot from 2016-11-05 17-36-28
screenshot from 2016-11-05 17-32-27

IMHO it makes better use of screen space and makes it simpler(not that it was very complex to begin with ;P)

**Comment by TingPing** [comment]
I think it looks worse, also continuing the accessibility talk this is a downgrade (Other than the placeholder text), as there is no discoverable way to close the window or add the station as the buttons previously provided.

**Comment by JasonLG1979** [comment]
I can add a widow title(it won't be visible but the window will have a title) . As far as buttons widow close button serves the same function as "Cancel" and "Add" isn't needed. When the treeview row is clicked the station is added and the dialog is closed(hidden).

**Comment by TingPing** [comment]
>  As far as buttons, the widow's close button serves the same function as "Cancel" and "Add" isn't needed. 

Well currently *Add* is actually impossible without a mouse. We need to bind activate or something so the keyboard can add. *Cancel* is only available via alt+f4 which is less than ideal and again not discoverable.

File: pithos/SearchDialog.py [hunk 1]
```diff
@@ -15,52 +15,59 @@
...
-    treeview = GtkTemplate.Child()
+    search_results_model = GtkTemplate.Child()
 
-    def __init__(self, *args, **kwargs):
-        self.worker_run = kwargs["worker"]
-        del kwargs["worker"]
-
-        super().__init__(*args, use_header_bar=1, **kwargs)
+    def __init__(self, worker_run):
+        super().__init__()
```
Comment (TingPing) [inline pithos/SearchDialog.py]: Always pass `**kwargs` to init, pygobject uses it for setting constructor properties.

File: pithos/StationsDialog.py [hunk 2]
```diff
@@ -38,7 +38,9 @@ def __init__(self, pithos, *args, **kwargs):
         self.model = pithos.stations_model
         self.worker_run = pithos.worker_run
         self.quickmix_changed = False
-        self.searchDialog = None
+        self.searchDialog = SearchDialog.SearchDialog(self.worker_run)
+        self.searchDialog.set_transient_for(self)
```
Comment (TingPing) [inline pithos/StationsDialog.py]: Once the `__init__()` func is fixed as mentioned above. `SearchDialog(self.worker_run, transient_for=self)`

Comment (JasonLG1979) [inline pithos/StationsDialog.py]: OK

**Comment by JasonLG1979** [comment]
> Well currently Add is actually impossible without a mouse. We need to bind activate or something so the keyboard can add. Cancel is only available via alt+f4 which is less than ideal and again not discoverable.

idk about "Add" but "Cancel" could just be removed and replaced with the window's close button. To me it's understood that closing a window means take no action which in this case would be the same as "Cancel".

File: pithos/SearchDialog.py [hunk 3]
```diff
@@ -15,52 +15,59 @@
...
-        self.result = None
+        self.worker_run = worker_run
+        self.query = ''
 
     @GtkTemplate.Callback
-    def search_clicked(self, widget):
-        self.search(self.entry.get_text())
+    def on_delete_event(self, *ignore):
+        self.query = ''
+        self.hide()
```
Comment (TingPing) [inline pithos/SearchDialog.py]: http://lazka.github.io/pgi-docs/index.html#Gtk-3.0/classes/Widget.html#Gtk.Widget.hide_on_delete

File: pithos/SearchDialog.py [hunk 4]
```diff
@@ -15,52 +15,59 @@
...
-            self.treeview.show()
-        self.worker_run('search', (query,), callback, "Searching...")
+                model.append((i.musicId, mk))
+        self.worker_run('search', (self.query,), callback, "Searching...")
 
-    def cursor_changed(self, *ignore):
-        self.result = self.get_selected()
-        self.set_response_sensitive(Gtk.ResponseType.OK, not not self.result)
+    @GtkTemplate.Callback
+    def on_treeview_button_press_event(self, treeview, event):
```
Comment (TingPing) [inline pithos/SearchDialog.py]: http://lazka.github.io/pgi-docs/index.html#Gtk-3.0/classes/TreeView.html#Gtk.TreeView.signals.row_activated

**Comment by JasonLG1979** [comment]
> I'd rather not keep repeating myself but I think for this specific change the previous design was better.

OK, what about just moving the search entry up into the headerbar to make better use of screen space?

**Comment by TingPing** [comment]
Why close this?

**Comment by JasonLG1979** [comment]
Because from now on all UI changes have to take accessibility into account and since we don't know what that actually entails I'm not going to waste my time on it.

**Comment by TingPing** [comment]
Well you did other changes that don't involve UI changes.

Also the guidelines to be accessible aren't that hard:

- Make everything navigable via keyboard
- If something isn't text, like icons, give it a description

My problems with this UI change was clear:

- You removed textual descriptions
- It is not navigable by keyboard

**Comment by JasonLG1979** [comment]
The only agreeable change was the placeholder text. I can put in a separate PR just for that.

**Pull request closed without merging**